pkg/apis/approvaltask/v1alpha1/approvaltask_validation.go

```diff
@@ -35,5 +37,123 @@ func (tg *ApprovalTask) Validate(ctx context.Context) *apis.FieldError {
 
 // Validate ApprovalTaskSpec
 func (tgs *ApprovalTaskSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
+	if tgs.NumberOfApprovalsRequired <= 0 {
+		errs = errs.Also(apis.ErrInvalidValue(tgs.NumberOfApprovalsRequired, "numberOfApprovalsRequired", "must be greater than 0"))
+	}
+
+	if len(tgs.Approvers) == 0 {
+		errs = errs.Also(apis.ErrMissingField("approvers"))
+	}
+
+	// Validate approvers
+	for i, approver := range tgs.Approvers {
+		fieldPath := fmt.Sprintf("approvers[%d]", i)
+		
+		if err := validateApprover(approver, fieldPath); err != nil {
+			errs = errs.Also(err)
+		}
+	}
+
+	// Note: We don't validate numberOfApprovalsRequired vs approver count because:
+	// 1. Groups can have multiple members (unknown at validation time)  
+	// 2. Group membership is resolved at runtime, not validation time
+	// 3. If there aren't enough approvers at runtime, the task stays "pending" (correct behavior)
+
+	return errs
+}
+
+// validateApprover validates a single approver entry
+func validateApprover(approver ApproverDetails, fieldPath string) *apis.FieldError {
+	var errs *apis.FieldError
+
+	// Validate approver name
+	if strings.TrimSpace(approver.Name) == "" {
+		errs = errs.Also(apis.ErrMissingField(fieldPath + ".name"))
+		return errs
+	}
+
+	// Validate approver type
+	approverType := DefaultedApproverType(approver.Type)
+	if approverType != "User" && approverType != "Group" {
+		errs = errs.Also(apis.ErrInvalidValue(approver.Type, fieldPath+".type", "must be either 'User' or 'Group'"))
+	}
+
+	// Validate name format based on type
+	if approverType == "User" {
+		if err := validateUserName(approver.Name); err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(approver.Name, fieldPath+".name", err.Error()))
+		}
+	} else if approverType == "Group" {
+		if err := validateGroupName(approver.Name); err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(approver.Name, fieldPath+".name", err.Error()))
+		}
+	}
+
+	// Validate input value
+	validInputs := []string{"pending", "approve", "reject"}
```

Comment:
There is another field partially approved. Could you check that? if it is required